### DISCUSSION TITLE
find daily total members, new members and communicator for a community server

### DISCUSSION THREAD

**Discussion by Varinderr** [Rapptz/discord.py#8087] [Q&A]
i am trying to find the discord stats for the community server where i am a moderator. it would be a great help if i can get some reference on how to begin with this project. Thank you!

**Comment by LostLuma** [comment]
You can use the [tasks extension](https://discordpy.readthedocs.io/en/stable/ext/tasks/index.html) to create a daily task that gathers these statistics for you.

The member count of a server can either be retrieved by fetching an invite belonging to the server, or if you have the `members` intent enabled you can simply check the `member_count` attribute on a given server as the library keeps it updated for you.

Discord does not expose the other two statistics to bots, so you will have to count these yourself over the course of the day. New members can be counted using the `on_member_join` event (requires `members` intent), and communicators can be counted using the `on_message` event (requires `guild_messages` intent).

**Reply by Varinderr** [reply]
thank you for your answer : )  i have member intent and guild message intent enabled. one question, in order to retrieve these details do i need a bot? and i am new with this, can you share any example for any of the attributes. thank you!

**Reply by LostLuma** [reply]
You need a bot in order to automatically do this, yes.
To retrieve the member count for an example guild or count communicators:
```py
client = discord.Client(...)
MY_SERVER = 272885620769161216

@tasks.loop(...)
async def collect_statistics():
    guild = client.get_guild(MY_SERVER)
    # Use guild.member_count and persist it

@client.event
async def on_message(message: discord.Message):
    if message.guild is None or message.guild.id != MY_SERVER:
        return

    # Use message.author to see which user wrote the message etc.
    # And add it to your counter to keep track of everyone's messages
```

You can see the attributes each object has in the documentation on the API reference page, see for example the [`Message` object](https://discordpy.readthedocs.io/en/stable/api.html#message).
